bug_1292234.patch

```diff
@@ -128,74 +128,80 @@ diff --git a/devtools/server/actors/storage.js b/devtools/server/actors/storage.
 +    }
 +  },
 +
-+  async populateStoreMap(host, storagePrincipal, storeMap) {
-+    const db = await ExtensionStorageIDB.open(storagePrincipal);
-+    this.dbConnectionForHost.set(host, db);
-+    const data = await db.get();
++  async getStoragePrincipal(extension) {
++    if (extension.getSharedData("storageIDBBackend")) {
++      return extension.getSharedData("storageIDBPrincipal");
++    }
++    // This extension either isn't using the IDB backend (false) or
++    // it hasn't migrated to the IDB backend yet (null)
 +
-+    for (const [key, value] of Object.entries(data)) {
-+      storeMap.set(key, value);
++    // Loop for web extension context related to the current window
++    let currentContext;
++    for (const view of extension.views) {
++      currentContext = view;
++    }
++    if (!currentContext) {
++      // We didn't find the extension context for the current target; give up.
++      return null;
 +    }
-+    return storeMap;
++    const {
++      backendEnabled,
++      storagePrincipal,
++    } = await ExtensionStorageIDB.selectBackend(currentContext);
++
++    if (!backendEnabled) {
++      // IDB backend disabled; give up.
++      return null;
++    }
++    return storagePrincipal;
 +  },
 +
 +  /**
 +  * This method asynchronously reads the browser.storage.local data for the window
 +  * (if it's an extension page window and the related extension has the storage
 +  * permission) and caches this data into this.hostVsStores.
 +  */
-+  async populateStoresForHost(host, window) {
++  async populateStoresForHost(host, window = this.storageActor.window) {
```

Comment:
I set this as a default value if it comes in as undefined instead of `const win = window || this.storageActor.window` in the method body

Though I could not find an instance where `window` was `undefined`, nor an instance where `this.storageActor.window` was `undefined` (and in these cases, `window === this.storageActor.window`.

The cases I tried with using my [test WE](https://github.com/biancadanforth/webextensions-examples/pull/3) were basically opening the storage inspector with and without a background script.

Also, we call this method from our storage actor's `preListStores` method, which currently always passes a `window` object.

All this is to say, I'm not sure we need to worry about it, but I guess we can just be extra sure by setting a default value -- what do you think?

Reply:
While the extension is being reloaded, we are making the webextension actor to "navigate" the devtools toolbox to a fallback window, so that we can keep the toolbox open while the entire extension is going away and then coming back once it has been started again:

https://searchfox.org/mozilla-central/rev/2c912888e3b7ae4baf161d98d7a01434f31830aa/devtools/server/actors/targets/webextension.js#154-168

and so I'm not sure if there is any possible scenario where we don't have any window at all.

Reply:
Hmm okay, yeah I can't think of one.

In fact, we don't even enter `populateStoresForHost` (the method where we are worried about an `undefined` value for `window`) when the extension is reloaded with the Storage Inspector opened.

In general, I found a related bug in this case, and I filed Issue #11 .